src/lib.rs

```diff
@@ -0,0 +1,969 @@
+        <BlogById<T>>::mutate(&blog_id, |inner_blog| inner_blog.unlock());
+        Self::deposit_event(RawEvent::BlogUnlocked(blog_owner, blog_id));
+        Ok(())
+    }
+
+    fn ensure_blog_exists(blog_id: T::BlogId) -> Result<Blog<T>, &'static str> {
+        ensure!(<BlogById<T>>::exists(blog_id), BLOG_NOT_FOUND);
+        Ok(Self::blog_by_id(blog_id))
+    }
+
+    fn ensure_post_exists(blog_id: T::BlogId, post_id: T::PostId) -> Result<Post<T>, &'static str> {
+        ensure!(<PostById<T>>::exists(blog_id, post_id), POST_NOT_FOUND);
+        Ok(Self::post_by_id(blog_id, post_id))
+    }
+
+    fn ensure_reply_exists(
+        blog_id: T::BlogId,
+        post_id: T::PostId,
+        reply_id: T::ReplyId,
+    ) -> Result<Reply<T>, &'static str> {
+        ensure!(
+            <ReplyById<T>>::exists((blog_id, post_id, reply_id)),
+            REPLY_NOT_FOUND
+        );
+        Ok(Self::reply_by_id((blog_id, post_id, reply_id)))
+    }
+
+    fn ensure_blog_ownership(
+        blog: &Blog<T>,
+        blog_owner: &T::BlogOwnerId,
+    ) -> Result<(), &'static str> {
+        ensure!(blog.is_owner(&blog_owner), BLOG_OWNERSHIP_ERROR);
+        Ok(())
+    }
+
+    fn ensure_reply_ownership(
+        reply: &Reply<T>,
+        reply_owner: &T::AccountId,
+    ) -> Result<(), &'static str> {
+        ensure!(reply.is_owner(&reply_owner), REPLY_OWNERSHIP_ERROR);
+        Ok(())
+    }
+
+    fn ensure_blog_unlocked(blog: &Blog<T>) -> Result<(), &'static str> {
+        ensure!(!blog.is_locked(), BLOG_LOCKED_ERROR);
+        Ok(())
+    }
+
+    fn ensure_post_unlocked(post: &Post<T>) -> Result<(), &'static str> {
+        ensure!(!post.is_locked(), POST_LOCKED_ERROR);
+        Ok(())
+    }
+
+    fn ensure_post_title_is_valid(title: &[u8]) -> Result<(), &'static str> {
+        ensure!(
+            title.len() <= T::PostTitleMaxLength::get() as usize,
+            POST_TITLE_TOO_LONG
+        );
+        Ok(())
+    }
+
+    fn ensure_post_body_is_valid(body: &[u8]) -> Result<(), &'static str> {
+        ensure!(
+            body.len() <= T::PostBodyMaxLength::get() as usize,
+            POST_BODY_TOO_LONG
+        );
+        Ok(())
+    }
+
+    fn ensure_posts_limit_not_reached(blog: &Blog<T>) -> Result<T::PostId, &'static str> {
+        // Get posts count, associated with given blog
+        let posts_count = blog.posts_count();
+
+        ensure!(
+            posts_count < T::PostsMaxNumber::get().into(),
+            POSTS_LIMIT_REACHED
+        );
+
+        Ok(posts_count)
+    }
+
+    /// Get either replies count or direct replies count by given parent post/blog.
+    pub fn get_replies_count(
+        blog_id: T::BlogId,
+        post_id: T::PostId,
+        reply_id: Option<T::ReplyId>,
+    ) -> usize {
+        // Calculate replies count, iterating through all post
+        // related replies and checking if reply parent is given parent post/reply
+        <ReplyById<T>>::enumerate()
```

Comment:
This kind of iteration is not safe in the runtime. This will iterate all replies that exist, and this is not only a set which will increase nearly monotonically, but an attacker could also strategically increase the set to break this iteration.
You will have to replace this computation, which normally is a very nice stateless approach, with having a counter field in the `Post` that you maintain over time.

Reply:
As a general rule, enumeration _inside of the runtime_ is only acceptable over sets that have a known upper bound on their size.

Reply:
Done https://github.com/iorveth/substrate-blog-module/commit/fa783699b67faf6de038d208ae594c0e1993c8e4#diff-b4aea3e418ccdb71239b96952d9cddb6L892